### PR TITLE
Convert tests to use Junit5

### LINKED ISSUE #74: Consider switching to Junit5
Per the short discussion on one of the recent PRS; there was a thought floated about switching to Junit5.

The Junit5 vintage plugin would be used to run the TestFX tests as I think they rely on the Junit4 API.

I'm opening this so the question can be discussed.

I'm my experience:
- Intellij can detect and run these tests absolutely fine.
- Gradle has awesome support including Gradle Script Kotlin staticly typed configuration support (added by yours truly in a PR against the Junit5 project earlier this month).
- The API is very clean and polished.
- Works fine with hamcrest assertions.

### PR DESCRIPTION
**Pull request by JLLeitschuh**
Closes #74

Also, I found this out while working on this:
https://github.com/TestFX/TestFX/issues/407

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
@AustinShalit How do I disable these PMD Junit4 checks? We aren't using Junit4 anymore.

**Comment by AustinShalit** [comment]
@JLLeitschuh I just pushed a commit to fix PMD.

Looks like there are still build errors.

**Comment by SamCarlberg** [comment]
@JLLeitschuh want to update to the api/app split?

**Comment by JLLeitschuh** [comment]
🤣  Bahahahahahahahah. 🤣 

Yea, I will. I love that merge conflict list.

**Comment by JLLeitschuh** [comment]
I also fixed the TestFX bug in this PR:
https://github.com/TestFX/TestFX/pull/409

File: app/app.gradle.kts [hunk 1]
```diff
@@ -40,30 +40,38 @@ tasks.withType<Jar> {
...
+ * Allows you to run the UI tests in headless mode by calling gradle with the -Pheadless argument
+ */
+if (project.hasProperty("jenkinsBuild") || project.hasProperty("headless")) {
+    println("Running UI Tests Headless")
+    junitPlatform {
+        filters {
+            tags {
+                /*
+                 * A category for UI tests that cannot run in headless mode, ie work properly with real windows
+                 * but not with the virtualized ones in headless mode.
```
Comment (SamCarlberg) [inline app/app.gradle.kts]: :+1:

File: app/src/test/java/edu/wpi/first/shuffleboard/app/json/TileSizeSaverTest.java [hunk 2]
```diff
@@ -31,9 +35,9 @@ public void deserialize() throws Exception {
     assertEquals(tileSize, new TileSize(1, 2));
   }
 
-  @Test(expected = JsonParseException.class)
+  @Test
   public void deserializeInvalid() throws Exception {
-    gson.fromJson("[1,-2]", TileSize.class);
+    assertThrows(JsonParseException.class, () -> gson.fromJson("[1,-2]", TileSize.class));
```
Comment (SamCarlberg) [inline app/src/test/java/edu/wpi/first/shuffleboard/app/json/TileSizeSaverTest.java]: `assertThrows` is really nice

File: build.gradle.kts [hunk 3]
```diff
@@ -142,41 +143,45 @@ task<Wrapper>("wrapper") {
 /**
  * Retrieves the [java][org.gradle.api.plugins.JavaPluginConvention] project convention.
  */
-val Project.`java`: org.gradle.api.plugins.JavaPluginConvention get() =
-    convention.getPluginByName("java")
+val Project.`java`: org.gradle.api.plugins.JavaPluginConvention
+    get() =
```
Comment (SamCarlberg) [inline build.gradle.kts]: Why the newline/whitespace changes here and later in the file?

Comment (JLLeitschuh) [inline build.gradle.kts]: Bad merge, I'll fix it.

**Pull request merged**